### DISCUSSION TITLE
New theme dark/light options seem to forget the default theme

### DISCUSSION THREAD

**Discussion by designbyadrian** [ghostty-org/ghostty#2763] [Q&A] [answered]
Testing the theme config options. Works well except some options go missing. On macOS, it's most apparent in the titlebar.

## Test config 1

```
palette = 0=#51576d
palette = 1=#e78284
palette = 2=#a6d189
palette = 3=#e5c890
palette = 4=#8caaee
palette = 5=#f4b8e4
palette = 6=#81c8be
palette = 7=#b5bfe2
palette = 8=#626880
palette = 9=#e78284
palette = 10=#a6d189
palette = 11=#e5c890
palette = 12=#8caaee
palette = 13=#f4b8e4
palette = 14=#81c8be
palette = 15=#a5adce
background = 303446
foreground = c6d0f5
cursor-color = f2d5cf
selection-background = 44495d
selection-foreground = c6d0f5
```

Result:

<img width="849" alt="image" src="https://github.com/user-attachments/assets/7033ea93-1cfb-4598-8e23-5ca3d92fca3f">

## Test config 2

```
theme = light:catppuccin-frappe,dark:catppuccin-macchiato
```

Result:

<img width="853" alt="image" src="https://github.com/user-attachments/assets/f7dcd0d5-3e8b-430b-96c8-e147495bedd2">

## Test config 3

```
theme = light:catppuccin-frappe,dark:catppuccin-macchiato

palette = 0=#51576d
palette = 1=#e78284
palette = 2=#a6d189
palette = 3=#e5c890
palette = 4=#8caaee
palette = 5=#f4b8e4
palette = 6=#81c8be
palette = 7=#b5bfe2
palette = 8=#626880
palette = 9=#e78284
palette = 10=#a6d189
palette = 11=#e5c890
palette = 12=#8caaee
palette = 13=#f4b8e4
palette = 14=#81c8be
palette = 15=#a5adce
background = 303446
foreground = c6d0f5
cursor-color = f2d5cf
selection-background = 44495d
selection-foreground = c6d0f5
```

Result:

<img width="859" alt="image" src="https://github.com/user-attachments/assets/44fde04a-6369-488c-a68e-abec1f8d5dd9">

**Comment by mitchellh** [comment]
I don’t know what I’m looking at. What’s missing or incorrect?

**Comment by designbyadrian** [comment]
The font colour in the titlebar. I assume it's a hint at other colours not coming through unless specified in the theme.

**Comment by mitchellh** [comment, marked as answer]
Oh. We don't set that, that's set by macOS depending on your system color theme. 

When you don't specify a separate light/dark theme, then `window-theme = auto` is the default. In this case, we force the window itself to use a system color scheme that matches your theme's background. For example, if your color theme is dark, we'll change the _window_ to macOS dark mode (while the rest of your system may still be light).

When you specify a light/dark theme, then `window-theme` becomes `system` (this is documented on `theme` by the way). This means we just use whatever your system theme is. And therefore macOS uses that color.

I was able to reproduce your screenshots but yes, its based on the system.

I think it's a reasonable request that our titlebar color for tabs and transparent should match our foreground color.  I'll open an issue for that.

**Reply by mitchellh** [reply]
#2764

**Reply by designbyadrian** [reply]
Thank you for understanding my issue, albeit small!
